backend/grievance_service.py

```
@@ -22,15 +22,22 @@ class GrievanceService:
     Main service for managing grievances, routing, and escalations.
     """
 
+    # Class-level cache to avoid redundant disk I/O when instantiating the service
+    _rules_cache = {}
```

Ruff (0.15.11)

[warning] Mutable default value for class attribute

(RUF012)